idaes/generic_models/costing/costing_base.py

```
@@ -0,0 +1,598 @@
+        cost_vars = DefaultCostingComponents
+        for v in cost_vars:
+            try:
+                cvar = getattr(self, v)
+                if not isinstance(cvar, pyo.Var):
+                    raise TypeError(
+                        f"{unit_model.name} {v} component must be a Var. "
+                        "Please check the costing package you are using to "
+                        "ensure that all costing components are declared as "
+                        "variables.")
+                elif cvar.lb is None or cvar.lb < 0:
+                    _log.warn(
+                        f"{unit_model.name} {v} component has a lower bound "
+                        "less than zero. Be aware that this may result in "
+                        "negative costs during optimization.")
+            except AttributeError:
+                pass
+
+    def initialize(self, *args, **kwargs):
+        """
+        Placeholder method for initialization
+        """
+        # TODO: Implement an initialization method
+        # TODO: Need to have a general pupose method (block triangularisation?)
+        # TODO: Should also allow registering custom methods
+
+        # Vars and Constraints
+        for c in DefaultCostingComponents:
+            if hasattr(self, c):
+                var = getattr(self, c)
+                cons = getattr(self, f"{c}_constraint")
+                calculate_variable_from_constraint(var, cons)
+
+    def del_costing(self):
+        # Need to clean up references before deletion
+
+        # Alias flowsheet costing block and unit model (parent)
+        fcb = self.config.flowsheet_costing_block
+
+        # Clean up references from fsb and unit_model
+        try:
+            fcb._registered_unit_costing.remove(self)
+        except ValueError:
+            # Assume this is OK
+            pass
+        try:
+            self.unit_model._initialization_order.remove(self)
+        except ValueError:
+            # Assume this is OK
```

Is this a good assumption?  I think we might want to not suppress this exception (this kind of thing has burned me in the past).

---

I think I fixed this.